File: src/main/java/com/medicalsystem/util/DateUtils.java
package com.medicalsystem.util;

import lombok.extern.java.Log;

import java.text.ParseException;
import java.time.LocalDate;
import java.time.ZoneId;
import java.util.Date;

@Log
public final class DateUtils {

    private static final String[] dateFormats = {
            "d.M.y",
            "M/d/y"
    };

    private DateUtils() {}

    public static LocalDate fromString(String str) throws ParseException {
        Date date = org.apache.commons.lang3.time.DateUtils.parseDateStrictly(str, dateFormats);
        return toLocalDate(date);
    }

    public static Date toUtilDate(LocalDate localDate) {
        if (localDate == null) {
            return null;
        }
        return Date.from(localDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
    }

    public static LocalDate toLocalDate(Date date) {
        return date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
    }
}

File: src/main/kotlin/com/medicalsystem/util/DateUtils.kt
package com.medicalsystem.util

import org.apache.commons.lang3.time.DateUtils
import java.text.SimpleDateFormat
import java.time.LocalDate
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.*

private val dateFormats = arrayOf(
        "yyyy-MM-dd",
        "d.M.y",
        "M/d/y"
)

private val jsonFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")

fun stringToLocalDate(s: String): LocalDate {
    val utilDate: Date = DateUtils.parseDateStrictly(s, *dateFormats)
    return utilToLocalDate(utilDate)
}

fun utilToLocalDate(utilDate: Date): LocalDate =
        utilDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDate()

fun LocalDate.toUtilDate(): Date =
        Date.from(this.atStartOfDay(ZoneId.systemDefault()).toInstant())

fun LocalDate.toJsonString(): String =
        this.format(jsonFormatter)